testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeActiveFilesCache.java

```diff
@@ -0,0 +1,107 @@
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.DATABRICKS_COMMUNICATION_FAILURE_MATCH;
+import static io.trino.tests.product.utils.QueryExecutors.onDelta;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+
+public class TestDeltaLakeActiveFilesCache
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Inject
+    @Named("s3.server_type")
+    private String s3ServerType;
+
+    private AmazonS3 s3;
+
+    @BeforeTestWithContext
+    public void setup()
+    {
+        super.setUp();
+        s3 = new S3ClientFactory().createS3Client(s3ServerType);
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testRefreshTheFilesCacheWhenTableIsRecreated()
+    {
+        String tableName = "test_dl_cached_table_files_refresh_" + randomNameSuffix();
+        String tableDirectory = "databricks-compatibility-test-" + tableName;
+
+        onTrino().executeQuery(format("CREATE TABLE delta.default.%s (col INT) WITH (location = 's3://%s/%s')",
+                tableName,
+                bucketName,
+                tableDirectory));
+
+        onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES 1");
+        // Add the files of the table in the active files cache
+        assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName)).containsOnly(row(1));
+
+        // Recreate the table outside Trino to avoid updating the Trino table active files cache
+        onDelta().executeQuery("DROP TABLE default." + tableName);
+        // Delete the contents of the table explicitly from storage (because it has been created as `EXTERNAL`)
+        deleteDirectory(tableDirectory);
+
+        onDelta().executeQuery(format("CREATE TABLE default.%s (col INTEGER) USING DELTA LOCATION 's3://%s/%s'",
+                tableName,
+                bucketName,
+                tableDirectory));
+        onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES 2");
+
+        // TODO https://github.com/trinodb/trino/issues/13737 Fix failure when active files cache is stale
+        assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
+                .hasMessageContaining("Error opening Hive split");
+
+        // Verify flushing cache resolve the query failure
+        onTrino().executeQuery("CALL delta.system.flush_metadata_cache(schema_name => 'default', table_name => '" + tableName + "')");
+        assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName)).containsOnly(row(2));
+
+        onTrino().executeQuery("DROP TABLE delta.default." + tableName);
+    }
+
+    private void deleteDirectory(String tableDirectory)
```

Comment:
We could extract and reuse `io.trino.tests.product.deltalake.TestDeltaLakeDatabricksCreateTableAsSelectCompatibility#removeS3Directory`